build/files.js

```diff
@@ -8,29 +8,78 @@
 
 module.exports['string_decoder.js'] = [
 
-    // pull in Bufer as a require
-    // add Buffer.isEncoding where missing
+    // we do not need internal/util anymore
     [
-        /^(\/\/ USE OR OTHER DEALINGS IN THE SOFTWARE\.)/m
-      ,   '$1\n\nvar Buffer = require(\'buffer\').Buffer;'
-        + '\n'
-        + '\nvar isBufferEncoding = Buffer.isEncoding'
+        /const internalUtil = require\('internal\/util'\);/
+      , ''
+    ]
+
+  , [
+        /const Buffer = require\('buffer'\).Buffer;/
+      , 'var Buffer = require(\'buffer\').Buffer;\n' +
+        'var bufferShim = require(\'buffer-shims\');'
+    ]
+
+    // allocUnsafe
+
+  , [
+      /Buffer\.((?:alloc)|(?:allocUnsafe)|(?:from))/g,
+      'bufferShim.$1'
+    ]
+
+    // add Buffer.isEncoding where missing
+  , [
+        /const isEncoding = Buffer\[internalUtil.kIsEncodingSymbol\];/
+      ,   '\nvar isEncoding = Buffer.isEncoding'
         + '\n  || function(encoding) {'
+        + '\n       encoding = \'\' + encoding'
         + '\n       switch (encoding && encoding.toLowerCase()) {'
         + '\n         case \'hex\': case \'utf8\': case \'utf-8\': case \'ascii\': case \'binary\': case \'base64\': case \'ucs2\': case \'ucs-2\': case \'utf16le\': case \'utf-16le\': case \'raw\': return true;'
         + '\n         default: return false;'
         + '\n       }'
         + '\n     }'
         + '\n'
 
+        + '\nfunction _normalizeEncoding(enc) {'
```

Comment:
if your doing it from scratch I highly recomend tempalte strings instead of this stuff, one million times easier to edit